---
title: 'Figure 3: Formal P-Values & Plots'
author: "Spiro Stilianoudakis"
date: "July 25, 2018"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Loading Libraries

```{r include=FALSE}
#base R packages
library(data.table)
library(dplyr)
library(geneplotter)
library(ggplot2)
#library(gplots)
library(RColorBrewer)
library(pheatmap)
library(ggbeeswarm)
library(rcompanion)
library(stringr)
library(openxlsx)

#bioconductor packages
#library(MultiAssayExperiment)
#library(GenomicRanges)
#library(IRanges)
#library(RaggedExperiment)
#library(SummarizedExperiment)

#source("http://bioconductor.org/biocLite.R")
#biocLite("edgeR")
#library(samr)
#library(edgeR)
#library(limma)
#library(DESeq2)
#library(EDASeq)
#library(hgu95av2.db)
#library(apeglm)
#library(genefilter)
#library(affy)
```

# Setting directory & importing workspace data

```{r}
setwd("C:/Users/Spiro Stilianoudakis/Documents/Microarray_methylation_materials/data")

load("preprocessed_CEL.Rdata")
load("preprocessed_data.Rdata")
```

# Methylation

## 2A. 

### Donor == "DDLT" & Injury == "low"

### comparison <- factor(sample_annotation$Type, levels = c("L1", "L2"))

```{r}
setwd("C:/Users/Spiro Stilianoudakis/Documents/Microarray_methylation_materials/results")

#set the index for subsetting the data according to the analysis of choice
ind <- (sample_annotation$Donor == "DDLT" & sample_annotation$Injury == "low")

#import methylation data
dems <- read.xlsx("2A.xlsx",
                  sheet = "DEGs",
                  startRow = 1,
                  colNames = TRUE)

#set cutoff to extract differentially methylated cpgs (value was obtained from file: /Methylation.Rmd line 487)
adj.p.cutoff <- 0.1
dems <- dems[as.numeric(dems$adj.P.Val) <= adj.p.cutoff,]
dim(dems)
#7265 


#using lfc to classify as hypomethylated or hypermethylated dm cpgs
dems$Hypomethylated <- ifelse(dems$logFC<0,1,0)
dems$Hypomethylated <- as.factor(dems$Hypomethylated)

table(dems$Hypomethylated)
#   0    1 
#3427 3838 

```

## 2A CGI Regions

```{r}
#CGI Regions
table(dems$Relation_to_UCSC_CpG_Island)

#regrouping cgi regions 
dems$Relation_to_UCSC_CpG_Island[grep("Shelf",dems$Relation_to_UCSC_CpG_Island)] <- "Shelf"
dems$Relation_to_UCSC_CpG_Island[grep("Shore",dems$Relation_to_UCSC_CpG_Island)] <- "Shore"
dems$Relation_to_UCSC_CpG_Island[grep("Island",dems$Relation_to_UCSC_CpG_Island)] <- "CGI"
dems$Relation_to_UCSC_CpG_Island[which(dems$Relation_to_UCSC_CpG_Island=="")] <- "Non_CGI"

dems$Relation_to_UCSC_CpG_Island <- as.factor(dems$Relation_to_UCSC_CpG_Island)

#sanity check
table(dems$Relation_to_UCSC_CpG_Island)

table(dems$Hypomethylated, dems$Relation_to_UCSC_CpG_Island)

#one way chisq. test for hypomethylated
chisq.test(table(dems$Relation_to_UCSC_CpG_Island[dems$Hypomethylated==1]))
#p-value < 2.2e-16


#overall chisq test type of methylation vs CGI
chisq.test(table(dems$Hypomethylated, dems$Relation_to_UCSC_CpG_Island))
#p-value < 2.2e-16

#pairwise chisq tests for methylation vs CGI at each level of CGI using bonferroni correction
pairwiseNominalIndependence(table(dems$Hypomethylated, dems$Relation_to_UCSC_CpG_Island), compare = "column",
                            fisher = FALSE,
                            gtest  = FALSE,
                            chisq  = TRUE,
                            method = "fdr")

dems$CGI <- ifelse(dems$Relation_to_UCSC_CpG_Island=="CGI",1,0)
dems$Non_CGI <- ifelse(dems$Relation_to_UCSC_CpG_Island=="Non_CGI",1,0)
dems$Shelf <- ifelse(dems$Relation_to_UCSC_CpG_Island=="Shelf",1,0)
dems$Shore <- ifelse(dems$Relation_to_UCSC_CpG_Island=="Shore",1,0)

chisq.test(dems$Hypomethylated, dems$CGI)$p.value
#0
chisq.test(dems$Hypomethylated, dems$Non_CGI)$p.value
#0
chisq.test(dems$Hypomethylated, dems$Shelf)$p.value
#6.256347e-119
chisq.test(dems$Hypomethylated, dems$Shore)$p.value
#3.193821e-17

#Bar plots
tab <- t(table(dems$Relation_to_UCSC_CpG_Island,dems$Hypomethylated))
dat <- data.frame(DiffMethy = c(rep("hyper",4),rep("hypo",4)), 
                  CGI = rep(c("CGI","Non_CGI","Shelf","Shore"),2),
                  Frequency = c(tab[1,],tab[2,]))

## stacked
ggplot(data=dat, aes(x=CGI, y=Frequency, fill=DiffMethy)) +
  geom_bar(stat="identity")

## grouped
ggplot(data=dat, aes(x=CGI, y=Frequency, fill=DiffMethy)) +
  geom_bar(stat="identity", position="dodge", color="black") + 
  xlab("CGI Region") + ggtitle("Low Injury (Pre-Imp vs Post-Rep)") +
  scale_fill_manual(values=c("grey", "black"), labels = c("Hypermethylated", "Hypomethylated")) +
  guides(fill=guide_legend(title="Methylation")) + scale_x_discrete(labels=c("CGI", "Non-CGI", "Shelf", "Shore")) +
  annotate("text", x=1, y=2500, label= "***", size=5) +
  annotate("text", x=2, y=1900, label= "***", size=5) +
  annotate("text", x=3, y=700, label= "***", size=5) +
  annotate("text", x=4, y=1150, label= "***", size=5) +
  annotate("text", x=5.5, y=2500, label= "Significance", size=4) +
  annotate("text", x=5.5, y=2300, label= "--   > 0.05", size=3.5) +
  annotate("text", x=5.5, y=2150, label= "*   < 0.01", size=3.5) +
  annotate("text", x=5.5, y=2000, label= "**   < 0.001", size=3.5) +
  annotate("text", x=5.5, y=1850, label= "***   < 0.0001", size=3.5) +
  coord_cartesian(xlim = c(1, 4), # This focuses the x-axis on the range of interest
                      clip = 'off' # This keeps the labels from disappearing
                      ) +
  theme_minimal()


#pie charts

#function to create blank theme
blank_theme <- theme_minimal()+
  theme(
  axis.title.x = element_blank(),
  axis.title.y = element_blank(),
  panel.border = element_blank(),
  panel.grid=element_blank(),
  axis.ticks = element_blank(),
  plot.title=element_text(size=14, face="bold")
  )

#hypomethylated cpgs
dat2 <- dat[dat$DiffMethy=="hypo",]
bp<- ggplot(dat2, aes(x="", y=Frequency, fill=CGI))+
geom_bar(width = 1, stat = "identity")
bp
prop.table(dat2$Frequency)
#0.62741011 0.07894737 0.01771756 0.27592496
pie <- bp + coord_polar("y", start=0) + scale_fill_discrete(name = "") + blank_theme + theme(axis.text.x=element_blank()) + theme(legend.text=element_text(size=20),legend.key.height=unit(2,"line"))
pie

#hypermethylated cpgs
dat2 <- dat[dat$DiffMethy=="hyper",]
bp<- ggplot(dat2, aes(x="", y=Frequency, fill=CGI))+
geom_bar(width = 1, stat = "identity")
bp
prop.table(dat2$Frequency)
#0.1126350 0.5202801 0.1756638 0.1914211
pie <- bp + coord_polar("y", start=0) + scale_fill_discrete(name = "") + blank_theme + theme(axis.text.x=element_blank()) + theme(legend.text=element_text(size=20),legend.key.height=unit(2,"line"))
pie

```

##2A Gene-Centric Regions

```{r}
#Gene-Centric Regions
table(dems$UCSC_RefGene_Group)

dems$UCSC_RefGene_Group[which(dems$UCSC_RefGene_Group=="")] <- "Non-Genic"

#list of the different categories
unique(unlist(strsplit(dems$UCSC_RefGene_Group,"\\|")))

#Compacting and Separating each level of the UCSC_RefGene_Group covariate
#using dplyr package
dems$UCSC_RefGene_Group <- sapply(dems$UCSC_RefGene_Group, function(x) strsplit(x, ";") %>% 
                                                  unlist %>% sort %>% unique %>% 
                                    paste(collapse="|")) %>% unlist %>% as.character


#separating the cpgs with multiple refgene groups
z <- dems[str_count(dems$UCSC_RefGene_Group, "\\|")>0,c(1,2,14)]
z <- z[order(z$UCSC_RefGene_Group),]

covariates <- matrix(nrow=dim(z)[1],ncol=length(unique(unlist(strsplit(z$UCSC_RefGene_Group,"\\|")))))
 
for(i in 1:length(unique(unlist(strsplit(z$UCSC_RefGene_Group,"\\|"))))){
         covariates[,i] <- ifelse(1:dim(z)[1] %in% grep(unique(unlist(strsplit(z$UCSC_RefGene_Group,"\\|")))[i],z$UCSC_RefGene_Group),1,0)
         
       }

covariates <- data.frame(covariates)
covariates$sum <- rowSums(covariates)

z2 <- cbind.data.frame(z,covariates)
colnames(z2)[dim(z)[2]+1:length(unique(unlist(strsplit(z$UCSC_RefGene_Group,"\\|"))))] <- unique(unlist(strsplit(z$UCSC_RefGene_Group,"\\|")))

z2 <- data.frame(z2[rep(seq_len(dim(z2)[1]), z2$sum),, drop = FALSE], row.names=NULL)

z2 <- z2 %>% group_by(ID) %>%
       mutate(num = 1:n())
z2 <- data.frame(z2)

z2$refgene <- NA

for(i in 1:dim(z2)[1]){
  z2$refgene[i] <- strsplit(z2$UCSC_RefGene_Group[i],"\\|")[[1]][z2$num[i]]
}

z2 <- z2[,c("ID","logFC", "refgene")]

#concatenate with the single group cpgs
y <- dems[str_count(dems$UCSC_RefGene_Group, "\\|")==0,c(1,2,14)]
colnames(y)[3] <- "refgene"

genecent <- rbind.data.frame(z2,y) 

table(genecent$refgene)

#grouping 5' UTR TSS1500 and TSS200 into Promotor category
genecent$refgene[which(genecent$refgene=="5UTR" |
                                  genecent$refgene=="TSS1500" |
                                  genecent$refgene=="TSS200")] <- "Promotor"

table(genecent$refgene)

genecent$Hypomethylated <- ifelse(genecent$logFC<0,1,0)

#one way chisq. test for hypomethylated
chisq.test(table(genecent$refgene[genecent$Hypomethylated==1]))
#p-value < 2.2e-16

#overall chisq test type of methylation vs Gene-Centric regions
chisq.test(table(genecent$Hypomethylated, genecent$refgene))
#p-value < 2.2e-16

#pairwise chisq tests of type of methylation vs Gene-Centric regions at each level of Gene-Centric regions using bonferroni correction
pairwiseNominalIndependence(table(genecent$Hypomethylated, genecent$refgene), compare = "column",
                            fisher = FALSE,
                            gtest  = FALSE,
                            chisq  = TRUE,
                            method = "bonferroni")

genecent$FirstExon <- ifelse(genecent$refgene=="1stExon",1,0)
genecent$ThreeUTR <- ifelse(genecent$refgene=="3UTR",1,0)
genecent$Body <- ifelse(genecent$refgene=="Body",1,0)
genecent$Non_Genic <- ifelse(genecent$refgene=="Non-Genic",1,0)
genecent$Promotor <- ifelse(genecent$refgene=="Promotor",1,0)

chisq.test(genecent$Hypomethylated, genecent$FirstExon)$p.value
#3.14513e-87
chisq.test(genecent$Hypomethylated, genecent$ThreeUTR)$p.value
#5.790246e-81
chisq.test(genecent$Hypomethylated, genecent$Body)$p.value
#2.347866e-229
chisq.test(genecent$Hypomethylated, genecent$Non_Genic)$p.value
#2.790058e-55
chisq.test(genecent$Hypomethylated, genecent$Promotor)$p.value
#8.010462e-301


#Barplots
tab <- t(table(genecent$refgene,genecent$Hypomethylated))
dat <- data.frame(DiffMethy = c(rep("hyper",5),rep("hypo",5)), 
                  "Gene Region" = rep(c("1st Exon", "3' UTR", "Body", "Non-Genic", "Promotor"),2),
                  Frequency = c(tab[1,],tab[2,]))

## stacked
ggplot(data=dat, aes(x=DiffMethy, y=Frequency, fill=Gene.Region)) +
  geom_bar(stat="identity")

## grouped
ggplot(data=dat, aes(x=Gene.Region, y=Frequency, fill=DiffMethy)) +
  geom_bar(stat="identity", position="dodge", color="black") + 
  xlab("Gene-Centric Region") + ggtitle("Low Injury (Pre-Imp vs Post-Rep)") +
  scale_fill_manual(values=c("grey", "black"), labels = c("Hypermethylated", "Hypomethylated")) +
  guides(fill=guide_legend(title="Methylation")) + scale_x_discrete(labels=c("1st Exon", "3' UTR", "Body", "Non-Genic", "Promotor")) +
  annotate("text", x=1, y=900, label= "***", size=5) +
  annotate("text", x=2, y=500, label= "***", size=5) +
  annotate("text", x=3, y=2000, label= "***", size=5) +
  annotate("text", x=4, y=900, label= "***", size=5) +
  annotate("text", x=5, y=3200, label= "***", size=5) +
  annotate("text", x=6.5, y=3000, label= "Significance", size=4) +
  annotate("text", x=6.5, y=2700, label= "--   > 0.05", size=3.5) +
  annotate("text", x=6.5, y=2550, label= "*   < 0.01", size=3.5) +
  annotate("text", x=6.5, y=2400, label= "**   < 0.001", size=3.5) +
  annotate("text", x=6.5, y=2250, label= "***   < 0.0001", size=3.5) +
  coord_cartesian(xlim = c(1, 5), 
                      clip = 'off') +
  theme_minimal()



#pie charts
#hypomethylated
dat2 <- dat[dat$DiffMethy=="hypo",]
bp<- ggplot(dat2, aes(x="", y=Frequency, fill=Gene.Region))+
geom_bar(width = 1, stat = "identity")
bp
prop.table(dat2$Frequency)
#0.141697274 0.009085637 0.174753528 0.077324570 0.597138991
pie <- bp + coord_polar("y", start=0) + scale_fill_discrete(name = "") + blank_theme + theme(axis.text.x=element_blank()) + theme(legend.text=element_text(size=20),legend.key.height=unit(2,"line"))
pie

#hypermethylated
dat2 <- dat[dat$DiffMethy=="hyper",]
bp<- ggplot(dat2, aes(x="", y=Frequency, fill=Gene.Region))+
geom_bar(width = 1, stat = "identity")
bp
prop.table(dat2$Frequency)
#0.01935484 0.09381720 0.49623656 0.18844086 0.20215054
pie <- bp + coord_polar("y", start=0) + scale_fill_discrete(name = "") + blank_theme + theme(axis.text.x=element_blank()) + theme(legend.text=element_text(size=20),legend.key.height=unit(2,"line"))
pie


```


## 2B. 

### Donor == "DDLT" & Injury == "high"

### comparison <- factor(sample_annotation$Type, levels = c("L1", "L2"))

```{r}

setwd("C:/Users/Spiro Stilianoudakis/Documents/Microarray_methylation_materials/results")

#set the index for subsetting the data according to the analysis of choice
ind <- (sample_annotation$Donor == "DDLT" & sample_annotation$Injury == "high")

#import methylation data
dems <- read.xlsx("2B.xlsx",
                  sheet = "DEGs",
                  startRow = 1,
                  colNames = TRUE)

#set cutoff to extract differentially methylated cpgs (value was obtained from file: /Methylation.Rmd line 487)
adj.p.cutoff <- 0.1
dems <- dems[as.numeric(dems$adj.P.Val) <= adj.p.cutoff,]
dim(dems)
#665 


#using lfc to classify as hypomethylated or hypermethylated dm cpgs
dems$Hypomethylated <- ifelse(dems$logFC<0,1,0)
dems$Hypomethylated <- as.factor(dems$Hypomethylated)

table(dems$Hypomethylated)
#  0   1 
# 94 571
```

##2B CGI Regions (steps the same as for 2A)

```{r}
table(dems$Relation_to_UCSC_CpG_Island)

#regrouping cgi regions
dems$Relation_to_UCSC_CpG_Island[grep("Shelf",dems$Relation_to_UCSC_CpG_Island)] <- "Shelf"
dems$Relation_to_UCSC_CpG_Island[grep("Shore",dems$Relation_to_UCSC_CpG_Island)] <- "Shore"
dems$Relation_to_UCSC_CpG_Island[grep("Island",dems$Relation_to_UCSC_CpG_Island)] <- "CGI"
dems$Relation_to_UCSC_CpG_Island[which(dems$Relation_to_UCSC_CpG_Island=="")] <- "Non_CGI"

dems$Relation_to_UCSC_CpG_Island <- as.factor(dems$Relation_to_UCSC_CpG_Island)

#sanity check
table(dems$Relation_to_UCSC_CpG_Island)

table(dems$Hypomethylated, dems$Relation_to_UCSC_CpG_Island)

#one way chisq. test for hypomethylated
chisq.test(table(dems$Relation_to_UCSC_CpG_Island[dems$Hypomethylated==1]))
#p-value < 2.2e-16

#overall chisq test type of methylation vs CGI
chisq.test(table(dems$Hypomethylated, dems$Relation_to_UCSC_CpG_Island))
#p-value < 2.2e-16

#pairwise chisq tests
pairwiseNominalIndependence(table(dems$Hypomethylated, dems$Relation_to_UCSC_CpG_Island), compare = "column",
                            fisher = FALSE,
                            gtest  = FALSE,
                            chisq  = TRUE,
                            method = "fdr")

dems$CGI <- ifelse(dems$Relation_to_UCSC_CpG_Island=="CGI",1,0)
dems$Non_CGI <- ifelse(dems$Relation_to_UCSC_CpG_Island=="Non_CGI",1,0)
dems$Shelf <- ifelse(dems$Relation_to_UCSC_CpG_Island=="Shelf",1,0)
dems$Shore <- ifelse(dems$Relation_to_UCSC_CpG_Island=="Shore",1,0)

chisq.test(dems$Hypomethylated, dems$CGI)$p.value
chisq.test(dems$Hypomethylated, dems$Non_CGI)$p.value
chisq.test(dems$Hypomethylated, dems$Shelf)$p.value
chisq.test(dems$Hypomethylated, dems$Shore)$p.value

#Bar plots
tab <- t(table(dems$Relation_to_UCSC_CpG_Island,dems$Hypomethylated))
dat <- data.frame(DiffMethy = c(rep("hyper",4),rep("hypo",4)), 
                  CGI = rep(c("CGI","Non_CGI","Shelf","Shore"),2),
                  Frequency = c(tab[1,],tab[2,]))

## stacked
ggplot(data=dat, aes(x=CGI, y=Frequency, fill=DiffMethy)) +
  geom_bar(stat="identity")

## grouped
ggplot(data=dat, aes(x=CGI, y=Frequency, fill=DiffMethy)) +
  geom_bar(stat="identity", position="dodge", color="black") + 
  xlab("CGI Region") + ggtitle("High Injury (Pre-Imp vs Post-Rep)") +
  scale_fill_manual(values=c("grey", "black"), labels = c("Hypermethylated", "Hypomethylated")) +
  guides(fill=guide_legend(title="Methylation")) + scale_x_discrete(labels=c("CGI", "Non-CGI", "Shelf", "Shore")) +
  annotate("text", x=1, y=60, label= "--", size=6) +
  annotate("text", x=2, y=300, label= "--", size=6) +
  annotate("text", x=3, y=100, label= "--", size=6) +
  annotate("text", x=4, y=160, label= "--", size=6) +
  annotate("text", x=5.5, y=275, label= "Significance", size=4) +
  annotate("text", x=5.5, y=250, label= "--   > 0.05", size=3.5) +
  annotate("text", x=5.5, y=235, label= "*   < 0.05", size=3.5) +
  annotate("text", x=5.5, y=220, label= "**   < 0.001", size=3.5) +
  annotate("text", x=5.5, y=205, label= "***   < 0.0001", size=3.5) +
  coord_cartesian(xlim = c(1, 4), # This focuses the x-axis on the range of interest
                      clip = 'off' # This keeps the labels from disappearing
                      ) +
  theme_minimal()


#pie charts
#hypo
dat2 <- dat[dat$DiffMethy=="hypo",]
bp<- ggplot(dat2, aes(x="", y=Frequency, fill=CGI))+
geom_bar(width = 1, stat = "identity")
bp
prop.table(dat2$Frequency)
#0.07880911 0.50788091 0.15761821 0.25569177
pie <- bp + coord_polar("y", start=0) + scale_fill_discrete(name = "") + blank_theme + theme(axis.text.x=element_blank()) + theme(legend.text=element_text(size=20),legend.key.height=unit(2,"line"))
pie

#hyper
dat2 <- dat[dat$DiffMethy=="hyper",]
bp<- ggplot(dat2, aes(x="", y=Frequency, fill=CGI))+
geom_bar(width = 1, stat = "identity")
bp
prop.table(dat2$Frequency)
#0.11702128 0.53191489 0.09574468 0.25531915
pie <- bp + coord_polar("y", start=0) + scale_fill_discrete(name = "") + blank_theme + theme(axis.text.x=element_blank()) + theme(legend.text=element_text(size=20),legend.key.height=unit(2,"line"))
pie

```

##2B Gene-Centric Regions (steps the same as for 2A)

```{r}
#Gene-Centric Regions
table(dems$UCSC_RefGene_Group)

dems$UCSC_RefGene_Group[which(dems$UCSC_RefGene_Group=="")] <- "Non-Genic"

#list of the different categories
unique(unlist(strsplit(dems$UCSC_RefGene_Group,"\\|")))

#Compacting and Separating each level of the UCSC_RefGene_Group covariate
#using dplyr package
dems$UCSC_RefGene_Group <- sapply(dems$UCSC_RefGene_Group, function(x) strsplit(x, ";") %>% 
                                                  unlist %>% sort %>% unique %>% 
                                    paste(collapse="|")) %>% unlist %>% as.character


#separating the cpgs with multiple refgene groups
z <- dems[str_count(dems$UCSC_RefGene_Group, "\\|")>0,c(1,2,14)]
z <- z[order(z$UCSC_RefGene_Group),]

covariates <- matrix(nrow=dim(z)[1],ncol=length(unique(unlist(strsplit(z$UCSC_RefGene_Group,"\\|")))))
 
for(i in 1:length(unique(unlist(strsplit(z$UCSC_RefGene_Group,"\\|"))))){
         covariates[,i] <- ifelse(1:dim(z)[1] %in% grep(unique(unlist(strsplit(z$UCSC_RefGene_Group,"\\|")))[i],z$UCSC_RefGene_Group),1,0)
         
       }

covariates <- data.frame(covariates)
covariates$sum <- rowSums(covariates)

z2 <- cbind.data.frame(z,covariates)
colnames(z2)[dim(z)[2]+1:length(unique(unlist(strsplit(z$UCSC_RefGene_Group,"\\|"))))] <- unique(unlist(strsplit(z$UCSC_RefGene_Group,"\\|")))

z2 <- data.frame(z2[rep(seq_len(dim(z2)[1]), z2$sum),, drop = FALSE], row.names=NULL)

z2 <- z2 %>% group_by(ID) %>%
       mutate(num = 1:n())
z2 <- data.frame(z2)

z2$refgene <- NA

for(i in 1:dim(z2)[1]){
  z2$refgene[i] <- strsplit(z2$UCSC_RefGene_Group[i],"\\|")[[1]][z2$num[i]]
}

z2 <- z2[,c("ID","logFC", "refgene")]

#concatenate with the single group cpgs
y <- dems[str_count(dems$UCSC_RefGene_Group, "\\|")==0,c(1,2,14)]
colnames(y)[3] <- "refgene"

genecent <- rbind.data.frame(z2,y) 

table(genecent$refgene)

genecent$refgene[which(genecent$refgene=="5UTR" |
                                  genecent$refgene=="TSS1500" |
                                  genecent$refgene=="TSS200")] <- "Promotor"
genecent$Hypomethylated <- ifelse(genecent$logFC<0,1,0)

#one way chisq. test for hypomethylated
chisq.test(table(genecent$refgene[genecent$Hypomethylated==1]))
#p-value < 2.2e-16

#overall chisq test type of methylation vs CGI
fisher.test(table(genecent$Hypomethylated, genecent$refgene))
#p-value = 0.2573

#pairwise chisq tests
pairwiseNominalIndependence(table(genecent$Hypomethylated, genecent$refgene), compare = "column",
                            fisher = TRUE,
                            gtest  = FALSE,
                            chisq  = FALSE,
                            method = "bonferroni")

tab <- t(table(genecent$refgene,genecent$Hypomethylated))

dat <- data.frame(DiffMethy = c(rep("hyper",5),rep("hypo",5)), 
                  "Gene Region" = rep(c("1st Exon", "3' UTR", "Body", "Non-Genic", "Promotor"),2),
                  Frequency = c(tab[1,],tab[2,]))

ggplot(data=dat, aes(x=DiffMethy, y=Frequency, fill=Gene.Region)) +
  geom_bar(stat="identity")

ggplot(data=dat, aes(x=DiffMethy, y=Frequency, fill=Gene.Region)) +
  geom_bar(stat="identity", position="dodge")

#pie charts
#hypo
dat2 <- dat[dat$DiffMethy=="hypo",]
bp<- ggplot(dat2, aes(x="", y=Frequency, fill=Gene.Region))+
geom_bar(width = 1, stat = "identity")
bp
prop.table(dat2$Frequency)
#0.0455259 0.0455259 0.5086342 0.1648352 0.2354788
pie <- bp + coord_polar("y", start=0) + scale_fill_discrete(name = "") + blank_theme + theme(axis.text.x=element_blank()) + theme(legend.text=element_text(size=20),legend.key.height=unit(2,"line"))
pie

#hyper
dat2 <- dat[dat$DiffMethy=="hyper",]
bp<- ggplot(dat2, aes(x="", y=Frequency, fill=Gene.Region))+
geom_bar(width = 1, stat = "identity")
bp
prop.table(dat2$Frequency)
#0.04504505 0.01801802 0.43243243 0.19819820 0.30630631
pie <- bp + coord_polar("y", start=0) + scale_fill_discrete(name = "") + blank_theme + theme(axis.text.x=element_blank()) + theme(legend.text=element_text(size=20),legend.key.height=unit(2,"line"))
pie

```

